### PR TITLE
refactor: use gomega for assertion in tests

### PR DESCRIPTION
**Pull request by chrisgacsal**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
Use `gomega` library for assertion in tests. Also remove use of `go-cmp` library.

### Why?
To align testing and dependencies with the `kafka-operator` project.

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/developer-guide/blob/master/docs/coding-style/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -116,6 +116,7 @@ github.com/alecthomas/units v0.0.0-20190717042225-c3de453c63f4/go.mod h1:ybxpYRF
 github.com/aliyun/alibaba-cloud-sdk-go v0.0.0-20190412020505-60e2075261b6/go.mod h1:T9M45xf79ahXVelWoOBmH0y4aC1t5kXO5BxwyakgIGA=
 github.com/aliyun/alibaba-cloud-sdk-go v0.0.0-20190620160927-9418d7b0cd0f h1:oRD16bhpKNAanfcDDVU+J0NXqsgHIvGbbe/sy+r6Rs0=
 github.com/aliyun/alibaba-cloud-sdk-go v0.0.0-20190620160927-9418d7b0cd0f/go.mod h1:myCDvQSzCW+wB1WAlocEru4wMGJxy+vlxHdhegi1CDQ=
+github.com/aliyun/alibaba-cloud-sdk-go v0.0.0-20190808125512-07798873deee h1:NYqDBPkhVYt68W3yoGoRRi32i3MLx2ey7SFkJ1v/UI0=
```
Comment (stoader) [inline go.sum:119]: do we need this? is this a transient dependency ?

File: properties/go.sum [hunk 2]
```diff
@@ -3,8 +3,29 @@ emperror.dev/errors v0.8.0/go.mod h1:YcRvLPh626Ubn2xqtoprejnA5nFha+TJ+2vew48kWuE
...
+github.com/golang/protobuf v1.2.0/go.mod h1:6lQm79b+lXiMfvg/cZm0SGofjICqVBUtrP5yJMmIC1U=
+github.com/golang/protobuf v1.4.0-rc.1/go.mod h1:ceaxUfeHdC40wWswd/P6IGgMaK3YpKi5j83Wpe3EHw8=
+github.com/golang/protobuf v1.4.0-rc.1.0.20200221234624-67d41d38c208/go.mod h1:xKAWHe0F5eneWXFV3EuXVDTCmh+JuBKY0li0aMyXATA=
+github.com/golang/protobuf v1.4.0-rc.2/go.mod h1:LlEzMj4AhA7rCAGe4KMBDvJI+AwstrUpVNzEA03Pprs=
+github.com/golang/protobuf v1.4.0-rc.4.0.20200313231945-b860323f09d0/go.mod h1:WU3c8KckQ9AFe+yFwt9sWVRKCVIyN9cPHBJSNnbL67w=
+github.com/golang/protobuf v1.4.0/go.mod h1:jodUvKwWbYaEsadDk5Fwe5c77LiNKVO9IDvqG2KuDX0=
+github.com/golang/protobuf v1.4.2 h1:+Z5KGCizgyZCbGh1KZqA0fcLLkwbsjIzS4aV2v7wJX0=
+github.com/golang/protobuf v1.4.2/go.mod h1:oDoupMAO8OvCJWAcko0GGGIgR6R6ocIYbsSw735rRwI=
+github.com/google/go-cmp v0.3.0/go.mod h1:8QqcDgzrUqlUb/G2PQTWiueGozuR1884gddMywk6iLU=
+github.com/google/go-cmp v0.3.1/go.mod h1:8QqcDgzrUqlUb/G2PQTWiueGozuR1884gddMywk6iLU=
```
Comment (stoader) [inline properties/go.sum:17]: looks like there’s still reference to go-cmp lib

**Review by stoader — CHANGES_REQUESTED** [review]

File: properties/go.sum (hunk 2 above)
Comment (chrisgacsal) [reply properties/go.sum:17]: Apparently yes. It is a transitive dependency of `gomega`.

File: go.sum (hunk 1 above)
Comment (chrisgacsal) [reply go.sum:119]: Yup. It looks like it:
```shell
$ go mod graph
...
github.com/go-acme/lego/v3@v3.1.0 github.com/aliyun/alibaba-cloud-sdk-go@v0.0.0-20190808125512-07798873deee
github.com/hashicorp/vault@v1.4.0 github.com/aliyun/alibaba-cloud-sdk-go@v0.0.0-20190620160927-9418d7b0cd0f
github.com/go-acme/lego/v3@v3.2.0 github.com/aliyun/alibaba-cloud-sdk-go@v0.0.0-20190808125512-07798873deee
github.com/hashicorp/vault-plugin-auth-alicloud@v0.5.4 github.com/aliyun/alibaba-cloud-sdk-go@v0.0.0-20190412020505-60e2075261b6
github.com/hashicorp/go-kms-wrapping@v0.0.0-20191129225826-634facde9f88 github.com/aliyun/alibaba-cloud-sdk-go@v0.0.0-20190620160927-9418d7b0cd0f
github.com/hashicorp/go-kms-wrapping@v0.5.1 github.com/aliyun/alibaba-cloud-sdk-go@v0.0.0-20190620160927-9418d7b0cd0f
github.com/hashicorp/vault-plugin-secrets-alicloud@v0.5.4 github.com/aliyun/alibaba-cloud-sdk-go@v0.0.0-20190620160927-9418d7b0cd0f
```

**Review by baluchicken — APPROVED** [review]

**Review by pbalogh-sa — APPROVED** [review]

**Review by stoader — APPROVED** [review]

File: properties/pkg/properties_test.go [hunk 3]
```diff
@@ -17,19 +17,18 @@ package properties
...
 func TestNewProperties(t *testing.T) {
-	p := NewProperties()
+	g := NewGomegaWithT(t)
 
-	if len(p.keys) != 0 && len(p.properties) != 0 {
-		t.Errorf("Newly created Properties is not empty")
-	}
+	p := NewProperties()
+	g.Expect(len(p.keys)).Should(Equal(0))
+	g.Expect(len(p.properties)).Should(Equal(0))
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:31]:
```suggestion
	g.Expect(p.keys).Should(BeEmpty())
	g.Expect(p.properties).Should(BeEmpty())
```

File: properties/internal/utils/struct_test.go [hunk 4]
```diff
@@ -17,132 +17,146 @@ package utils
...
 		expected := &structTagFlag{
 			Key: "omitempty",
 		}
+
 		stf, err := parseStructTagFlag("omitempty")
 
-		if err != nil {
-			t.Errorf("Parsing a non-empty struc tag flag string should not return an error!")
-		}
+		g.Expect(err).Should(Succeed(),
```
Comment (hi-im-aren) [inline properties/internal/utils/struct_test.go:109]: In many places I saw this `g.Expect(err).Should(Succeed())` expression, I think the original intention with Succeed() is to use it in pair with an `Expect(FUNCTION())`. 

I would suggest using `g.Expect(err).Should(BeNil())`, so it's more expressive for go developers.

File: properties/pkg/properties_test.go [hunk 5]
```diff
@@ -593,13 +562,13 @@ func TestProperties_Equal(t *testing.T) {
 			nextKeyIndex: 3,
 		}
 
-		if p.Equal(&expected) {
-			t.Errorf("Comparing Properties two objects with different set of keys must not be equal!")
-		}
+		g.Expect(p.Equal(expected)).ShouldNot(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:565]:
```suggestion
		g.Expect(p).ShouldNot(BeEquivalentTo(expected))
```

File: properties/pkg/properties_test.go [hunk 6]
```diff
@@ -625,15 +594,15 @@ func TestProperties_Equal(t *testing.T) {
 			nextKeyIndex: 3,
 		}
 
-		if p.Equal(&expected) {
-			t.Errorf("Comparing Properties two objects with same set of keys, but different Property objects must not be equal!")
-		}
+		g.Expect(p.Equal(expected)).Should(BeFalse())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:597]:
```suggestion
		g.Expect(p).ShouldNot(BeEquivalentTo(expected))
```

File: properties/pkg/properties_test.go [hunk 7]
```diff
@@ -664,13 +633,13 @@ func TestProperties_Sort(t *testing.T) {
 
 		p.Sort()
 
-		if !cmp.Equal(p, expected) {
-			t.Errorf("Mismatch in expected and returned Properties!\nExpected: %q\nGot: %q\n",
-				expected, p)
-		}
+		g.Expect(p.Keys()).Should(Equal(expected.Keys()))
+		g.Expect(p.Equal(expected)).Should(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:637]:
```suggestion
		g.Expect(p).Should(BeEquivalentTo(expected))
```
Not sure the keys() check is required, I think the deepEqual under the hood could in fact cover it.

File: properties/pkg/properties_test.go [hunk 8]
```diff
@@ -701,10 +670,8 @@ func TestProperties_Sort(t *testing.T) {
...
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:674]:
```suggestion
		g.Expect(p).Should(BeEquivalentTo(expected))
```

File: properties/pkg/properties_test.go [hunk 9]
```diff
@@ -564,13 +533,13 @@ func TestProperties_Equal(t *testing.T) {
 			nextKeyIndex: 2,
 		}
 
-		if p.Equal(&expected) {
-			t.Errorf("Comparing Properties two objects with different set of keys must not be equal!")
-		}
+		g.Expect(p.Equal(expected)).ShouldNot(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:536]:
```suggestion
		g.Expect(p).ShouldNot(BeEquivalentTo(expected))
```

File: properties/pkg/properties_test.go [hunk 10]
```diff
@@ -540,13 +509,13 @@ func TestProperties_Equal(t *testing.T) {
 			nextKeyIndex: 3,
 		}
 
-		if p.Equal(&expected) {
-			t.Errorf("Comparing valid and malformed Properties must not be equal!")
-		}
+		g.Expect(p.Equal(expected)).ShouldNot(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:512]:
```suggestion
		g.Expect(p).ShouldNot(BeEquivalentTo(expected))
```

File: properties/pkg/properties_test.go [hunk 11]
```diff
@@ -485,26 +454,25 @@ func TestProperties_MergeDefaults(t *testing.T) {
...
+			nextKeyIndex: 2,
 		}
 
 		dst.MergeDefaults(src)
-
-		if !dst.Equal(expectedProperties) {
-			t.Errorf("Mismatch in expected and returned Properties!\nExpected: %q\nGot: %q\n",
-				expectedProperties, dst)
-		}
+		g.Expect(dst.Equal(expectedProperties)).Should(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:475]:
```suggestion
		g.Expect(dst).Should(BeEquivalentTo(expectedProperties))
```

File: properties/pkg/properties_test.go [hunk 12]
```diff
@@ -515,13 +483,14 @@ func TestProperties_Equal(t *testing.T) {
 	p.put(Property{"test.key3", "test.value3", "this is a comment line"})
 
 	t.Run("Nil", func(t *testing.T) {
-		if p.Equal(nil) {
-			t.Errorf("Comparing valid and malformed Properties must not be equal!")
-		}
+		g := NewGomegaWithT(t)
+		g.Expect(p.Equal(nil)).Should(BeFalse())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:487]:
```suggestion
		g.Expect(p).ShouldNot(BeNil())
```

File: properties/pkg/properties_test.go [hunk 13]
```diff
@@ -446,34 +413,36 @@ func TestProperties_MergeDefaults(t *testing.T) {
...
+			nextKeyIndex: 3,
 		}
 
 		dst.MergeDefaults(src)
-
-		if !dst.Equal(expectedProperties) {
-			t.Errorf("Mismatch in expected and returned Properties!\nExpected: %q\nGot: %q\n",
-				expectedProperties, dst)
-		}
+		g.Expect(dst.Equal(expectedProperties)).Should(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:440]:
```suggestion
		g.Expect(dst).Should(BeEquivalentTo(expectedProperties))
```

File: properties/pkg/properties_test.go [hunk 14]
```diff
@@ -407,34 +372,36 @@ func TestProperties_MergeDefaults(t *testing.T) {
...
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:399]:
```suggestion
		g.Expect(dst).Should(BeEquivalentTo(expectedProperties))
```

File: properties/pkg/properties_test.go [hunk 15]
```diff
@@ -283,120 +247,121 @@ func TestProperties_Len(t *testing.T) {
...
+			nextKeyIndex: 2,
 		}
 
 		dst.MergeDefaults(nil)
-
-		if !dst.Equal(expectedProperties) {
-			t.Errorf("Mismatch in expected and returned Properties!\nExpected: %q\nGot: %q\n",
-				expectedProperties, dst)
-		}
+		g.Expect(dst.Equal(expectedProperties)).Should(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:359]:
```suggestion
		g.Expect(dst).Should(BeEquivalentTo(expectedProperties))
```

File: properties/pkg/properties_test.go [hunk 16]
```diff
@@ -283,120 +247,121 @@ func TestProperties_Len(t *testing.T) {
...
-		if !p1.Equal(&expectedProperties) {
-			t.Errorf("Mismatch in expected and returned Properties!\nExpected: %q\nGot: %q\n",
-				&expectedProperties, p1)
-		}
+		src := NewProperties()
+		src.put(Property{"test.key2", "p2", "this is a comment line"})
+		src.put(Property{"test.key3", "p2", "this is a comment line"})
+
+		dst.Merge(src)
+		g.Expect(dst.Equal(expectedProperties)).Should(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:325]:
```suggestion
		g.Expect(dst).Should(BeEquivalentTo(expectedProperties))
```

File: properties/pkg/properties_test.go [hunk 17]
```diff
@@ -283,120 +247,121 @@ func TestProperties_Len(t *testing.T) {
...
+			nextKeyIndex: 2,
 		}
 
 		dst.Merge(nil)
-
-		if !dst.Equal(expectedProperties) {
-			t.Errorf("Mismatch in expected and returned Properties!\nExpected: %q\nGot: %q\n",
-				expectedProperties, dst)
-		}
+		g.Expect(dst.Equal(expectedProperties)).Should(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:284]:
```suggestion
		g.Expect(dst).Should(BeEquivalentTo(expectedProperties))
```

File: properties/pkg/properties_test.go [hunk 18]
```diff
@@ -189,75 +174,54 @@ func TestProperties_SetWithComment(t *testing.T) {
...
-
-		if !found {
-			t.Errorf("Newly added Property is not found.")
-		}
-
-		if !prop.Equal(expectedProperty) {
-			t.Errorf("Mismatch between return and expected list of keys. Expected: %v, got %v", prop, expectedProperty)
-		}
+		g.Expect(found).Should(BeTrue())
+		g.Expect(prop.Equal(expectedProperty)).Should(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:224]:
```suggestion
		g.Expect(prop).Should(BeEquivalentTo(expectedProperty))
```

File: properties/pkg/properties_test.go [hunk 19]
```diff
@@ -189,75 +174,54 @@ func TestProperties_SetWithComment(t *testing.T) {
...
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:207]:
```suggestion
		g.Expect(prop).Should(BeEquivalentTo(expectedProperty))
```

File: properties/pkg/properties_test.go [hunk 20]
```diff
@@ -189,75 +174,54 @@ func TestProperties_SetWithComment(t *testing.T) {
...
-
-		if !found {
-			t.Error("Newly added Property is not found.")
-		}
-
-		if !prop.Equal(expectedProperty) {
-			t.Errorf("Mismatch between return and expected list of keys. Expected: %q, got %q", prop, expectedProperty)
-		}
+		g.Expect(found).Should(BeTrue())
+		g.Expect(prop.Equal(expectedProperty)).Should(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:190]:
```suggestion
		g.Expect(prop).Should(BeEquivalentTo(expectedProperty))
```

File: properties/pkg/properties_test.go [hunk 21]
```diff
@@ -115,72 +121,51 @@ func TestProperties_Set(t *testing.T) {
...
-
-		if !found {
-			t.Errorf("Newly added Property is not found.")
-		}
-
-		if !prop.Equal(expectedProperty) {
-			t.Errorf("Mismatch between return and expected list of keys. Expected: %v, got %v", prop, expectedProperty)
-		}
+		g.Expect(found).Should(BeTrue())
+		g.Expect(prop.Equal(expectedProperty)).Should(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:168]:
```suggestion
		g.Expect(prop).Should(BeEquivalentTo(expectedProperty))
```

File: properties/pkg/properties_test.go [hunk 22]
```diff
@@ -115,72 +121,51 @@ func TestProperties_Set(t *testing.T) {
...
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:152]:
```suggestion
		g.Expect(prop).Should(BeEquivalentTo(expectedProperty))
```

File: properties/pkg/properties_test.go [hunk 23]
```diff
@@ -115,72 +121,51 @@ func TestProperties_Set(t *testing.T) {
...
-
-		if !found {
-			t.Error("Newly added Property is not found.")
-		}
-
-		if !prop.Equal(expectedProperty) {
-			t.Errorf("Mismatch between return and expected list of keys. Expected: %q, got %q", prop, expectedProperty)
-		}
+		g.Expect(found).Should(BeTrue())
+		g.Expect(prop.Equal(expectedProperty)).Should(BeTrue())
```
Comment (hi-im-aren) [inline properties/pkg/properties_test.go:136]:
```suggestion
		g.Expect(prop).Should(BeEquivalentTo(expectedProperty))
```

File: properties/pkg/properties_test.go (hunk 5 above)
Comment (chrisgacsal) [reply properties/pkg/properties_test.go:565]: I'd not change this as here we need to test the correctness of `Equal` method which does not take the `comment` field of `Property` struct into the consideration when comparing two `Properties` objects.

File: properties/pkg/properties_test.go (hunk 6 above)
Comment (chrisgacsal) [reply properties/pkg/properties_test.go:597]: Same as above.

File: properties/pkg/properties_test.go (hunk 9 above)
Comment (chrisgacsal) [reply properties/pkg/properties_test.go:536]: Same as above!

File: properties/pkg/properties_test.go (hunk 11 above)
Comment (chrisgacsal) [reply properties/pkg/properties_test.go:475]: Same as above.

File: properties/pkg/properties_test.go (hunk 10 above)
Comment (chrisgacsal) [reply properties/pkg/properties_test.go:512]: Same above.

**Comment by chrisgacsal** [comment]
@hi-im-aren Thanks a lot for your review. I have added the changes you suggested in the following PR: https://github.com/banzaicloud/kafka-operator/pull/559

**Pull request merged**